---
title: "createAllTablesAndFigures"
output: html_document
date: "2024-09-02"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABIAAAASCAYAAABWzo5XAAAAWElEQVR42mNgGPTAxsZmJsVqQApgmGw1yApwKcQiT7phRBuCzzCSDSHGMKINIeDNmWQlA2IigKJwIssQkHdINgxfmBBtGDEBS3KCxBc7pMQgMYE5c/AXPwAwSX4lV3pTWwAAAABJRU5ErkJggg==
---

# Preamble
```{r libraries}
library(tidyverse)
library(ggplot2)
library(ggpubr)
library(ggrepel)

library(vctrs)
library(scales)
library(rTensor)
library(cluster)
library(factoextra)
library(RColorBrewer)

#remotes::install_github("coolbutuseless/ggpattern")
library(ggpattern)
```

```{r define colours}

# Old approach
# Needed colours: BMI groups (3) and phyla levels (6)
# colours = hue_pal()(9)
# #show_col(colours)
# BMI_cols = colours[c(1,4,7)]
# phylum_cols = colours[-c(1,4,7)]

# Rasmus's suggestion
colours = RColorBrewer:: brewer.pal(8, "Dark2")
BMI_cols = c("darkgreen","darkgoldenrod","darkred") #colours[1:3]
WHZ_cols = c("darkgreen", "darkgoldenrod", "dodgerblue4")
phylum_cols = colours#[-c(1:3)]
```

```{r import raw data}
faeces_taxa = read.csv("./newTaxonomy_faeces.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(faeces_taxa) = c("OTU", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
milk_taxa = read.csv("./newTaxonomy_milk.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(milk_taxa) = c("OTU", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
milkMetab_features = read.csv("./milk_metab_CAS_numbers.csv") %>% as_tibble()
colnames(milkMetab_features) = c("Metabolite", "CAS")

faeces_raw = read.csv("./faecesCounts.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(faeces_raw) = faeces_taxa$OTU
milk_raw = read.csv("./milkCounts.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(milk_raw) = milk_taxa$OTU
milkMetab_raw = read.csv("./milkMetabNumeric.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(milkMetab_raw) = milkMetab_features$Metabolite
```

```{r import BMI NPLS models}
faeces_path = "./20240110_faeces_NPLS_BMI/"
milk_path = "./20240110_milk_NPLS_BMI/"
milkMetab_path = "./20240110_milkMetab_NPLS_BMI/"

faeces_BMI_A = read.csv(paste0(faeces_path, "Rasmus_mode1.csv"), header=FALSE) %>% as_tibble()
colnames(faeces_BMI_A) = c("Component_1", "subject", "WeightGroup")
faeces_BMI_B = read.csv(paste0(faeces_path, "Rasmus_mode2.csv"), header=FALSE) %>% as_tibble()
colnames(faeces_BMI_B) = c("Component_1", "OTU", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
faeces_BMI_C = read.csv(paste0(faeces_path, "Rasmus_mode3.csv"), header=FALSE) %>% as_tibble()
colnames(faeces_BMI_C) = c("Component_1", "Timepoint")
faeces_BMI_modelled = read.csv(paste0(faeces_path, "Rasmus_model.csv"), header=FALSE) %>% as_tibble()
faeces_BMI_modelled = k_fold(as.matrix(faeces_BMI_modelled), 1, c(159, 93, 3))
faeces_BMI_input = read.csv(paste0(faeces_path, "Rasmus_input.csv"), header=FALSE) %>% as_tibble()
faeces_BMI_input = k_fold(as.matrix(faeces_BMI_input), 1, c(159, 93, 3))

# Flip for Rasmus's plots later on
# faeces_BMI_A$Component_1 = -1 * faeces_BMI_A$Component_1
# faeces_BMI_B$Component_1 = -1 * faeces_BMI_B$Component_1

milk_BMI_A = read.csv(paste0(milk_path, "Rasmus_mode1.csv"), header=FALSE) %>% as_tibble()
colnames(milk_BMI_A) = c("Component_1", "subject", "WeightGroup")
milk_BMI_B = read.csv(paste0(milk_path, "Rasmus_mode2.csv"), header=FALSE) %>% as_tibble()
colnames(milk_BMI_B) = c("Component_1", "OTU", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
milk_BMI_C = read.csv(paste0(milk_path, "Rasmus_mode3.csv"), header=FALSE) %>% as_tibble()
colnames(milk_BMI_C) = c("Component_1", "Timepoint")
milk_BMI_modelled = read.csv(paste0(milk_path, "Rasmus_model.csv"), header=FALSE) %>% as_tibble()
milk_BMI_modelled = k_fold(as.matrix(milk_BMI_modelled), 1, c(169, 115, 4))
milk_BMI_input = read.csv(paste0(milk_path, "Rasmus_input.csv"), header=FALSE) %>% as_tibble()
milk_BMI_input = k_fold(as.matrix(milk_BMI_input), 1, c(169, 115, 4))

# Flip for Rasmus's plots later on
# milk_BMI_A$Component_1 = -1 * milk_BMI_A$Component_1
# milk_BMI_B$Component_1 = -1 * milk_BMI_B$Component_1

milkMetab_BMI_A = read.csv(paste0(milkMetab_path, "Rasmus_mode1.csv"), header=FALSE) %>% as_tibble()
colnames(milkMetab_BMI_A) = c("Component_1", "subject", "WeightGroup")
milkMetab_BMI_B = read.csv(paste0(milkMetab_path, "Rasmus_mode2.csv"), header=FALSE) %>% as_tibble()
colnames(milkMetab_BMI_B) = c("Component_1", "Metabolite", "CAS")
milkMetab_BMI_C = read.csv(paste0(milkMetab_path, "Rasmus_mode3.csv"), header=FALSE) %>% as_tibble()
colnames(milkMetab_BMI_C) = c("Component_1", "Timepoint")
milkMetab_BMI_modelled = read.csv(paste0(milkMetab_path, "Rasmus_model.csv"), header=FALSE) %>% as_tibble()
milkMetab_BMI_modelled = k_fold(as.matrix(milkMetab_BMI_modelled), 1, c(164, 70, 4))
milkMetab_BMI_input = read.csv(paste0(milkMetab_path, "Rasmus_input.csv"), header=FALSE) %>% as_tibble()
milkMetab_BMI_input = k_fold(as.matrix(milkMetab_BMI_input), 1, c(164, 70, 4))

# Flip for Rasmus's plots later on
#milkMetab_BMI_A$Component_1 = -1 * milkMetab_BMI_A$Component_1
#milkMetab_BMI_B$Component_1 = -1 * milkMetab_BMI_B$Component_1

# Map categories to metabolites
metaboliteCategories = read.csv("./20240906_data/Metabolite_categories.txt", sep="\t") %>% as_tibble()
metaboliteCategories = metaboliteCategories %>% mutate(Metabolite = vec_as_names(Metabolite, repair="universal_quiet"))
milkMetab_BMI_B = milkMetab_BMI_B %>% left_join(metaboliteCategories)

## Add mismatches by hand
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Aminobutyrate","Class"] = "Amino acids and derivatives"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Fucosyllactose","Class"] = "Oligosaccharides"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Hydroxybutyrate","Class"] = "Amino acids and derivatives"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Oxoglutarate","Class"] = "Energy related"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X3.Fucosyllactose","Class"] = "Oligosaccharides"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X3SL.partial","Class"] = "Oligosaccharides"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X6SL.partial","Class"] = "Oligosaccharides"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Methionine","Class"] = "Amino acids and derivatives"
```

```{r import WHZ 6M NPLS models}
faeces_path = "./20240110_faeces_NPLS_WHZ_6m/"
milk_path = "./20240110_milk_NPLS_WHZ_6m/"
milkMetab_path = "./20240110_milkMetab_NPLS_WHZ_6m/"

faeces_WHZ_A = read.csv(paste0(faeces_path, "Rasmus_mode1.csv"), header=FALSE) %>% as_tibble()
colnames(faeces_WHZ_A) = c("Component_1", "subject", "WeightGroup")
faeces_WHZ_B = read.csv(paste0(faeces_path, "Rasmus_mode2.csv"), header=FALSE) %>% as_tibble()
colnames(faeces_WHZ_B) = c("Component_1", "OTU", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
faeces_WHZ_C = read.csv(paste0(faeces_path, "Rasmus_mode3.csv"), header=FALSE) %>% as_tibble()
colnames(faeces_WHZ_C) = c("Component_1", "Timepoint")
faeces_WHZ_modelled = read.csv(paste0(faeces_path, "Rasmus_model.csv"), header=FALSE) %>% as_tibble()
faeces_WHZ_modelled = k_fold(as.matrix(faeces_WHZ_modelled), 1, c(132, 93, 3))
faeces_WHZ_input = read.csv(paste0(faeces_path, "Rasmus_input.csv"), header=FALSE) %>% as_tibble()
faeces_WHZ_input = k_fold(as.matrix(faeces_WHZ_input), 1, c(132, 93, 3))

# Flip for Rasmus's plots later on
# faeces_WHZ_A$Component_1 = -1 * faeces_WHZ_A$Component_1
# faeces_WHZ_B$Component_1 = -1 * faeces_WHZ_B$Component_1

milk_WHZ_A = read.csv(paste0(milk_path, "Rasmus_mode1.csv"), header=FALSE) %>% as_tibble()
colnames(milk_WHZ_A) = c("Component_1", "subject", "WeightGroup")
milk_WHZ_B = read.csv(paste0(milk_path, "Rasmus_mode2.csv"), header=FALSE) %>% as_tibble()
colnames(milk_WHZ_B) = c("Component_1", "OTU", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
milk_WHZ_C = read.csv(paste0(milk_path, "Rasmus_mode3.csv"), header=FALSE) %>% as_tibble()
colnames(milk_WHZ_C) = c("Component_1", "Timepoint")
milk_WHZ_modelled = read.csv(paste0(milk_path, "Rasmus_model.csv"), header=FALSE) %>% as_tibble()
milk_WHZ_modelled = k_fold(as.matrix(milk_WHZ_modelled), 1, c(135, 115, 4))
milk_WHZ_input = read.csv(paste0(milk_path, "Rasmus_input.csv"), header=FALSE) %>% as_tibble()
milk_WHZ_input = k_fold(as.matrix(milk_WHZ_input), 1, c(135, 115, 4))

# Flip for Rasmus's plots later on
# milk_WHZ_A$Component_1 = -1 * milk_WHZ_A$Component_1
# milk_WHZ_B$Component_1 = -1 * milk_WHZ_B$Component_1

milkMetab_WHZ_A = read.csv(paste0(milkMetab_path, "Rasmus_mode1.csv"), header=FALSE) %>% as_tibble()
colnames(milkMetab_WHZ_A) = c("Component_1", "subject", "WeightGroup")
milkMetab_WHZ_B = read.csv(paste0(milkMetab_path, "Rasmus_mode2.csv"), header=FALSE) %>% as_tibble()
colnames(milkMetab_WHZ_B) = c("Component_1", "Metabolite", "CAS")
milkMetab_WHZ_C = read.csv(paste0(milkMetab_path, "Rasmus_mode3.csv"), header=FALSE) %>% as_tibble()
colnames(milkMetab_WHZ_C) = c("Component_1", "Timepoint")
milkMetab_WHZ_modelled = read.csv(paste0(milkMetab_path, "Rasmus_model.csv"), header=FALSE) %>% as_tibble()
milkMetab_WHZ_modelled = k_fold(as.matrix(milkMetab_WHZ_modelled), 1, c(136, 70, 4))
milkMetab_WHZ_input = read.csv(paste0(milkMetab_path, "Rasmus_input.csv"), header=FALSE) %>% as_tibble()
milkMetab_WHZ_input = k_fold(as.matrix(milkMetab_WHZ_input), 1, c(136, 70, 4))

# Flip for Rasmus's plots later on
milkMetab_WHZ_A$Component_1 = -1 * milkMetab_WHZ_A$Component_1
milkMetab_WHZ_B$Component_1 = -1 * milkMetab_WHZ_B$Component_1

# Add categories
milkMetab_WHZ_B = milkMetab_WHZ_B %>% left_join(milkMetab_BMI_B %>% select(-Component_1,-CAS)) 
```
```{r repair metabolomics names}
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Aminobutyrate","Metabolite"] = "2-Aminobutyrate"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Fucosyllactose","Metabolite"] = "2-Fucosyllactose"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Hydroxybutyrate","Metabolite"] = "2-Hydroxybutyrate"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X2.Oxoglutarate","Metabolite"] = "2-Oxoglutarate"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X3.Fucosyllactose","Metabolite"] = "3-Fucosyllactose"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X3SL.partial","Metabolite"] = "3SL partial"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "X6SL.partial","Metabolite"] = "6SL partial"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.1.70.Sialylated","Metabolite"] = "Unk. HMO 1.70 Sialylated"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.5.076","Metabolite"] = "Unk. HMO 5.076"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.5.09","Metabolite"] = "Unk. HMO 5.09"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.5.10","Metabolite"] = "Unk. HMO 5.10"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.5.11","Metabolite"] = "Unk. HMO 5.11"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.5.29","Metabolite"] = "Unk. HMO 5.29"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "Unk..HMO.5.359","Metabolite"] = "Unk. HMO 5.359"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "sn.Glycero.3.phosphocholine","Metabolite"] = "sn-Glycero-3-phosphocholine"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "myo.Inositol","Metabolite"] = "myo-Inositol"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "cis.Aconitate","Metabolite"] = "cis-Aconitate"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "DSLNT.partial","Metabolite"] = "DSLNT partial"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "O.Phosphocholine","Metabolite"] = "O-Phosphocholine"
milkMetab_BMI_B[milkMetab_BMI_B$Metabolite == "O.Acetylcarnitine","Metabolite"] = "O-Acetylcarnitine"

milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X2.Aminobutyrate","Metabolite"] = "2-Aminobutyrate"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X2.Fucosyllactose","Metabolite"] = "2-Fucosyllactose"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X2.Hydroxybutyrate","Metabolite"] = "2-Hydroxybutyrate"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X2.Oxoglutarate","Metabolite"] = "2-Oxoglutarate"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X3.Fucosyllactose","Metabolite"] = "3-Fucosyllactose"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X3SL.partial","Metabolite"] = "3SL partial"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "X6SL.partial","Metabolite"] = "6SL partial"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.1.70.Sialylated","Metabolite"] = "Unk. HMO 1.70 Sialylated"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.5.076","Metabolite"] = "Unk. HMO 5.076"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.5.09","Metabolite"] = "Unk. HMO 5.09"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.5.10","Metabolite"] = "Unk. HMO 5.10"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.5.11","Metabolite"] = "Unk. HMO 5.11"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.5.29","Metabolite"] = "Unk. HMO 5.29"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "Unk..HMO.5.359","Metabolite"] = "Unk. HMO 5.359"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "sn.Glycero.3.phosphocholine","Metabolite"] = "sn-Glycero-3-phosphocholine"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "myo.Inositol","Metabolite"] = "myo-Inositol"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "cis.Aconitate","Metabolite"] = "cis-Aconitate"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "DSLNT.partial","Metabolite"] = "DSLNT partial"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "O.Phosphocholine","Metabolite"] = "O-Phosphocholine"
milkMetab_WHZ_B[milkMetab_WHZ_B$Metabolite == "O.Acetylcarnitine","Metabolite"] = "O-Acetylcarnitine"
```

# Check the meaning of the loadings and clusters with Y
```{r comparison with y}
# 11/12/2024: all loadings are correct, no further flipping needed
# Positive loading = positive association with ppBMI/WHZ
# Negative loading = negative association with ppBMI/WHZ

faeces_BMI_y = read.csv("./20240110_faeces_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% select(V4) %>% pull()
faeces_BMI_cor = cor(faeces_BMI_modelled[,,1]@data, faeces_BMI_y)
cbind(faeces_BMI_B, faeces_BMI_cor) %>% as_tibble() %>% arrange(desc(Component_1)) 

milk_BMI_y = read.csv("./20240110_milk_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% select(V4) %>% pull()
milk_BMI_cor = cor(milk_BMI_modelled[,,1]@data, milk_BMI_y)
cbind(milk_BMI_B, milk_BMI_cor) %>% as_tibble() %>% arrange(desc(Component_1)) 

milkMetab_BMI_y = read.csv("./20240110_milkMetab_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% select(V4) %>% pull()
milkMetab_BMI_cor = cor(milkMetab_BMI_modelled[,,1]@data, milkMetab_BMI_y)
cbind(milkMetab_BMI_B, milkMetab_BMI_cor) %>% as_tibble() %>% arrange(desc(Component_1)) 

faeces_WHZ_y = read.csv("./20240110_faeces_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% select(V4) %>% pull()
faeces_WHZ_cor = cor(faeces_WHZ_modelled[,,1]@data, faeces_WHZ_y)
cbind(faeces_WHZ_B, faeces_WHZ_cor) %>% as_tibble() %>% arrange(desc(Component_1)) 

milk_WHZ_y = read.csv("./20240110_milk_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% select(V4) %>% pull()
milk_WHZ_cor = cor(milk_WHZ_modelled[,,1]@data, milk_WHZ_y)
cbind(milk_WHZ_B, milk_WHZ_cor) %>% as_tibble() %>% arrange(desc(Component_1)) 

milkMetab_WHZ_y = read.csv("./20240110_milkMetab_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% select(V4) %>% pull()
milkMetab_WHZ_cor = cor(milkMetab_WHZ_modelled[,,1]@data, milkMetab_WHZ_y)
cbind(milkMetab_WHZ_B, milkMetab_WHZ_cor) %>% as_tibble() %>% arrange(desc(Component_1)) 
```

# Mother ppBMI
```{r mother ppbmi - Supplementary Figure Sx}
df = read.csv("./20241031_data/faeces_asv_meta.csv") %>% as_tibble() %>% mutate(subject = str_split_fixed(RCID, "_", 2)[,1])

df %>% select(subject, ppBMI, BMI.group) %>% unique() %>% mutate(BMI.group = factor(BMI.group, levels=c("Normal", "Overweight", "Obese"))) %>% ggplot(aes(x=ppBMI,fill=as.factor(BMI.group))) + geom_histogram(bins=35,col="black") + scale_fill_manual(name="ppBMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols)
```

# Infant weight
```{r infant weight}
weights = read.csv("./20241107_data_w_anthropometrics/faeces_asv_meta.csv") %>% as_tibble() %>% mutate(subject = str_split_fixed(RCID, "_", 2)[,1])
weights = weights %>% select(subject, BMI.group, Birth.weight, Weight.month.1, Weight.month.2, Weight.month.3, Weight.month.6, Weight.1y, Weight.2y, Weight.3y) %>% unique() %>% pivot_longer(-c(subject,BMI.group))
weights = weights %>% as.data.frame()

weights[(weights$value >= 1000) & !is.na(weights$value),"value"] = weights[(weights$value >= 1000) & !is.na(weights$value),"value"] / 1000
weights$month = 0
weights[weights$name == "Birth.weight", "month"] = 0
weights[weights$name == "Weight.month.1", "month"] = 1
weights[weights$name == "Weight.month.2", "month"] = 2
weights[weights$name == "Weight.month.3", "month"] = 3
weights[weights$name == "Weight.month.6", "month"] = 6
weights[weights$name == "Weight.1y", "month"] = 12
weights[weights$name == "Weight.2y", "month"] = 24
weights[weights$name == "Weight.3y", "month"] = 36

# Plot 1
weights %>% filter(value != "NA") %>% ggplot(aes(x=as.factor(month),y=value,fill=as.factor(BMI.group))) + geom_boxplot() + geom_point(position=position_dodge(width=0.75)) + scale_fill_manual(name="ppBMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + xlab("Time post-partum [months]") + ylab("Infant weight [kg]")

weights %>% filter(value != "NA") %>% ggplot(aes(x=as.factor(month),y=value,fill=as.factor(BMI.group))) + facet_wrap(~BMI.group) + geom_boxplot() + geom_line(aes(group=as.factor(subject)), col="gray") + geom_point() + scale_fill_manual(name="ppBMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + xlab("Time post-partum [months]") + ylab("Infant weight [kg]")



whz_weights = read.csv("../../Data_for_Roel_w_Anthropometrics/Data_for_Roel_w_Anthropometrics/faeces_asv_meta.csv") %>% as_tibble() %>% mutate(subject = str_split_fixed(RCID, "_", 2)[,1]) %>% select(subject, BMI.group, whz.6m, whz.1y, whz.2y, whz.3y) %>% unique() %>% pivot_longer(-c(subject,BMI.group)) %>% filter(value != "NA")

whz_groups = whz_weights %>% filter(name == "whz.6m") %>% select(-name,-BMI.group) %>% unique() %>% mutate(whz_group = "Normal", subject = as.numeric(subject))
whz_groups[whz_groups$value <= -1, "whz_group"] = "Underweight"
whz_groups[whz_groups$value >= 1, "whz_group"] = "Overweight"
whz_groups %>% ggplot(aes(x=value,fill=whz_group)) + geom_histogram(bins=25, col="black")
whz_groups = whz_groups %>% select(subject, whz_group)

whz_weights = whz_weights %>% as.data.frame()

whz_weights$month = NA
whz_weights[whz_weights$name == "whz.6m", "month"] = 6
whz_weights[whz_weights$name == "whz.1y", "month"] = 12
whz_weights[whz_weights$name == "whz.2y", "month"] = 24
whz_weights[whz_weights$name == "whz.3y", "month"] = 36

bmi_weights = read.csv("./20240906_data/faeces_asv_meta.csv") %>% as_tibble() %>% mutate(subject = str_split_fixed(RCID, "_", 2)[,1]) %>% select(subject, BMI.group, BMIz.1y, BMIz.2y, BMIz.3y) %>% unique() %>% pivot_longer(-c(subject,BMI.group))
bmi_weights = bmi_weights %>% as.data.frame()

bmi_weights$month = NA
bmi_weights[bmi_weights$name == "BMIz.1y", "month"] = 12
bmi_weights[bmi_weights$name == "BMIz.2y", "month"] = 24
bmi_weights[bmi_weights$name == "BMIz.3y", "month"] = 36

rbind(whz_weights, bmi_weights) %>% as_tibble() %>% filter(name %in% c("whz.6m", "whz.1y", "BMIz.2y", "BMIz.3y")) %>% mutate(name = factor(name, levels=c("whz.6m", "whz.1y", "BMIz.2y", "BMIz.3y"))) %>% ggplot(aes(x=name,y=value,fill=as.factor(BMI.group))) + geom_boxplot() + scale_fill_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + scale_x_discrete(labels=c("6 months [WHZ]", "12 months [WHZ]", "24 months [BMIz]", "36 months [BMIz]")) + xlab("Time point [metric]") + ylab("Z-score") + theme(legend.position="na")

rbind(whz_weights, bmi_weights) %>% as_tibble() %>% filter(name %in% c("whz.6m", "whz.1y", "BMIz.2y", "BMIz.3y")) %>% mutate(name = factor(name, levels=c("whz.6m", "whz.1y", "BMIz.2y", "BMIz.3y"))) %>% ggplot(aes(x=name,y=value)) + facet_wrap(~BMI.group) + geom_boxplot(aes(fill=as.factor(BMI.group))) + geom_line(aes(group=as.factor(subject)),col="grey") + geom_point() + scale_fill_manual(name="ppBMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + scale_x_discrete(labels=c("6 months [WHZ]", "12 months [WHZ]", "24 months [BMIz]", "36 months [BMIz]")) + xlab("Time point [metric]") + ylab("Z-score") + theme(legend.position="na") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

#Export weight bocplot
weight.boxplot <- rbind(whz_weights, bmi_weights) %>% as_tibble() %>% filter(name %in% c("whz.6m", "whz.1y", "BMIz.2y", "BMIz.3y")) %>% mutate(name = factor(name, levels=c("whz.6m", "whz.1y", "BMIz.2y", "BMIz.3y"))) %>% ggplot(aes(x=name,y=value,fill=as.factor(BMI.group))) + geom_boxplot() + scale_fill_manual(name="ppBMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + scale_x_discrete(labels=c("6 months [WHZ]", "12 months [WHZ]", "24 months [BMIz]", "36 months [BMIz]")) + xlab("Time point [metric]") + ylab("Infant growth [Z-score]")

save(weight.boxplot,file="weight_boxplot.rds")
```
# NPLS model plots

```{r figure S2A faecal npls bmi}
a = faeces_BMI_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + theme(legend.position = "none") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

b = faeces_BMI_B %>% arrange(Kingdom, Phylum, Class, Order, Family, Genus, Species, OTU) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(Phylum))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) + theme(legend.position = "none") + ggtitle("Feature mode")

c = faeces_BMI_C %>% ggplot(aes(x=Timepoint,y=Component_1)) + geom_point() + geom_line() + xlab("Time point [days]") + ylab("") + ggtitle("Time mode")

plot = ggarrange(a,b,c,nrow=1)
annotate_figure(plot, top = text_grob("Infant faecal microbiome NPLS BMI"))

# Fake for label
faeces_BMI_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="ppBMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

rbind(faeces_BMI_B, faeces_WHZ_B, milk_BMI_B, milk_WHZ_B) %>% ggplot(aes(x=1:nrow(.), y=Component_1, fill=as.factor(Phylum))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols)
```

```{r figure S2B milk npls bmi}
a = milk_BMI_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + theme(legend.position = "none") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

b = milk_BMI_B %>% arrange(Kingdom, Phylum, Class, Order, Family, Genus, Species, OTU) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(Phylum))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols) + theme(legend.position = "none") + ggtitle("Feature mode")

c = milk_BMI_C %>% ggplot(aes(x=Timepoint,y=Component_1)) + geom_point() + geom_line() + xlab("Time point [days]") + ylab("") + ggtitle("Time mode")

plot = ggarrange(a,b,c,nrow=1)
annotate_figure(plot, top = text_grob("Mother milk microbiome NPLS BMI"))
```

```{r figure S2C milk metab npls bmi}
a = milkMetab_BMI_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + theme(legend.position = "none") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

b = milkMetab_BMI_B %>% arrange(Class,Metabolite) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(Class))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Metabolite class", labels=c("Amino acids and derivatives", "Energy related", "Fatty acids and derivatives", "Food derived", "Microbially derived", "Oligosaccharides", "Pharmaceuticals", "Sugars"), values=phylum_cols) + theme(legend.position = "none") + ggtitle("Feature mode")

c = milkMetab_BMI_C %>% ggplot(aes(x=Timepoint,y=Component_1)) + geom_point() + geom_line() + xlab("Time point [days]") + ylab("") + ggtitle("Time mode")

plot = ggarrange(a,b,c,nrow=1)
annotate_figure(plot, top = text_grob("Mother milk metabolome NPLS BMI"))
```

```{r figure S3A faecal npls whz}
a = faeces_WHZ_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + theme(legend.position = "none") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="WHZ group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

b = faeces_WHZ_B %>% arrange(Kingdom, Phylum, Class, Order, Family, Genus, Species, OTU) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(Phylum))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) + theme(legend.position = "none") + ggtitle("Feature mode")

c = faeces_WHZ_C %>% ggplot(aes(x=Timepoint,y=Component_1)) + geom_point() + geom_line() + xlab("Time point [days]") + ylab("") + ggtitle("Time mode")

plot = ggarrange(a,b,c,nrow=1)
annotate_figure(plot, top = text_grob("Infant faecal microbiome NPLS WHZ 6M"))
```

```{r figure S3B milk npls whz}
a = milk_WHZ_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + theme(legend.position = "none") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

b = milk_WHZ_B %>% arrange(Kingdom, Phylum, Class, Order, Family, Genus, Species, OTU) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(Phylum))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols) + theme(legend.position = "none") + ggtitle("Feature mode")

c = milk_WHZ_C %>% ggplot(aes(x=Timepoint,y=Component_1)) + geom_point() + geom_line() + xlab("Time point [days]") + ylab("") + ggtitle("Time mode")

plot = ggarrange(a,b,c,nrow=1)
annotate_figure(plot, top = text_grob("Mother milk microbiome NPLS WHZ 6M"))
```

```{r figure S3C milk metab npls whz}
a = milkMetab_WHZ_A %>% arrange(WeightGroup,subject) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(WeightGroup))) + geom_bar(stat="identity") + theme(legend.position = "none") + xlab("Subject index") + ylab("Component 1") + scale_fill_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + ggtitle("Subject mode")

b = milkMetab_WHZ_B %>% arrange(Class,Metabolite) %>% mutate(index=1:nrow(.)) %>% ggplot(aes(x=index,y=Component_1,fill=as.factor(Class))) + geom_bar(stat="identity") + xlab("Feature index") + ylab("") + scale_fill_manual(name="Metabolite class", labels=c("Amino acids and derivatives", "Energy related", "Fatty acids and derivatives", "Food derived", "Microbially derived", "Oligosaccharides", "Pharmaceuticals", "Sugars"), values=phylum_cols) + theme(legend.position = "none") + ggtitle("Feature mode")

c = milkMetab_WHZ_C %>% ggplot(aes(x=Timepoint,y=Component_1)) + geom_point() + geom_line() + xlab("Time point [days]") + ylab("") + ggtitle("Time mode")

plot = ggarrange(a,b,c,nrow=1)
annotate_figure(plot, top = text_grob("Mother milk metabolome NPLS WHZ 6M"))
```

# Figure 2 alternative
```{r Fig 2 alt metabolomics loadings}

# Faeces BMI
temp = faeces_BMI_B %>% 
  arrange(Component_1) %>% 
  mutate(Genus = gsub("g__", "", Genus), Species = gsub("s__", "", Species)) %>% 
  mutate(Species = str_split_fixed(Species, "_", 2)[,2]) %>% 
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "sp."), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "organism"), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "bacterium"), "", .x)))

temp[temp$Species == "", "Species"] = "sp."
temp$name = paste0(temp$Genus, " ", temp$Species)
temp[temp$name == " sp.", "name"] = "Unclassified"
temp = temp %>% filter(abs(Component_1) >= 0.1) %>% arrange(Component_1) %>% mutate(index=1:nrow(.))

e=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria"), values=phylum_cols[-3]) + theme(legend.position="none", axis.text.y = element_text(face="italic")) + ylab("")

# Milk BMI
temp = milk_BMI_B %>% 
  arrange(Component_1) %>% 
  mutate(Genus = gsub("g__", "", Genus), Species = gsub("s__", "", Species)) %>% 
  mutate(Species = str_split_fixed(Species, "_", 2)[,2]) %>% 
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "sp."), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "organism"), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "bacterium"), "", .x)))

temp[temp$Species == "", "Species"] = "sp."
temp$name = paste0(temp$Genus, " ", temp$Species)
temp[temp$name == " sp.", "name"] = "Unclassified"
temp = temp %>% filter(abs(Component_1) >= 0.1) %>% arrange(Component_1) %>% mutate(index=1:nrow(.))

d=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) + theme(legend.position="none", axis.text.y = element_text(face="italic")) + ylab("")

# Milk metabolomics BMI
temp = milkMetab_BMI_B %>% arrange(Component_1) %>% filter(abs(Component_1) >= 0.1) %>% mutate(index=1:nrow(.))
c = temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Class))) + geom_bar_pattern(stat="identity", pattern_angle=45, pattern_density=0.05, pattern_spacing=0.02, pattern_key_scale_factor=0.6, col="black",pattern_fill="black") + ylab("Metabolite") + xlab("Component 1") + scale_y_discrete(labels=temp$Metabolite) + scale_fill_manual(name="Metabolite class", labels=c("Amino acids and derivatives", "Energy related", "Fatty acids and derivatives", "Food derived", "Microbially derived", "Oligosaccharides", "Pharmaceuticals", "Sugars"), values=phylum_cols) + theme(legend.position="none") + ylab("")

# Faecal WHZ
temp = faeces_WHZ_B %>% 
  arrange(Component_1) %>% 
  mutate(Genus = gsub("g__", "", Genus), Species = gsub("s__", "", Species)) %>% 
  mutate(Species = str_split_fixed(Species, "_", 2)[,2]) %>% 
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "sp."), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "organism"), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "bacterium"), "", .x)))

temp[temp$Species == "", "Species"] = "sp."
temp$name = paste0(temp$Genus, " ", temp$Species)
temp[temp$name == " sp.", "name"] = "Unclassified"
temp = temp %>% filter(abs(Component_1) >= 0.1) %>% arrange(Component_1) %>% mutate(index=1:nrow(.))

h=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) + theme(legend.position="none", axis.text.y = element_text(face="italic")) + ylab("")

# Milk WHZ
temp = milk_WHZ_B %>% 
  arrange(Component_1) %>% 
  mutate(Genus = gsub("g__", "", Genus), Species = gsub("s__", "", Species)) %>% 
  mutate(Species = str_split_fixed(Species, "_", 2)[,2]) %>% 
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "sp."), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "organism"), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "bacterium"), "", .x)))

temp[temp$Species == "", "Species"] = "sp."
temp$name = paste0(temp$Genus, " ", temp$Species)
temp[temp$name == " sp.", "name"] = "Unclassified"
temp = temp %>% filter(abs(Component_1) >= 0.1) %>% arrange(Component_1) %>% mutate(index=1:nrow(.))

g=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols) + theme(legend.position="none", axis.text.y = element_text(face="italic")) + ylab("")

# Milk Metabolomics WHZ
temp = milkMetab_WHZ_B %>% arrange(Component_1) %>% filter(abs(Component_1) >= 0.1) %>% mutate(index=1:nrow(.))
f = temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Class))) + geom_bar_pattern(stat="identity", pattern_angle=45, pattern_density=0.05, pattern_spacing=0.02, pattern_key_scale_factor=0.6, col="black", pattern_fill="black") + ylab("Metabolite") + xlab("Component 1") + scale_y_discrete(labels=temp$Metabolite) + scale_fill_manual(name="Metabolite class", labels=c("Amino acids and derivatives", "Energy related", "Fatty acids and derivatives", "Food derived", "Microbially derived", "Oligosaccharides", "Pharmaceuticals", "Sugars"), values=phylum_cols) + theme(legend.position="none") + ylab("")

c
d
e
f
g
h
```
```{r suppl sx loadings of all models unfiltered}
temp=faeces_BMI_B %>% arrange(Component_1) %>% mutate(name=paste0(Genus,";",Species)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>% mutate(index=1:nrow(.))
a=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria"), values=phylum_cols[-3]) + theme(legend.position="none") + ylab("")

temp=milk_BMI_B %>% arrange(Component_1) %>% mutate(name=paste0(Genus,";",Species)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>% mutate(index=1:nrow(.))
temp[temp$name == "", "name"] = "Unclassified"
b=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobecteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) + theme(legend.position="none") + ylab("")

temp = milkMetab_BMI_B %>% arrange(Component_1) %>% mutate(index=1:nrow(.))
c = temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Class))) + geom_bar_pattern(stat="identity", pattern_angle=45, pattern_density=0.05, pattern_spacing=0.02, pattern_key_scale_factor=0.6, col="black",pattern_fill="black") + ylab("Metabolite") + xlab("Component 1") + scale_y_discrete(labels=temp$Metabolite) + scale_fill_manual(name="Metabolite class", labels=c("Amino acids and derivatives", "Energy related", "Fatty acids and derivatives", "Food derived", "Microbially derived", "Oligosaccharides", "Pharmaceuticals", "Sugars"), values=phylum_cols) + theme(legend.position="none") + ylab("")

temp=faeces_WHZ_B %>% arrange(Component_1) %>% mutate(name=paste0(Genus,";",Species)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>% mutate(index=1:nrow(.))
d=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria"), values=phylum_cols[-3]) + theme(legend.position="none") + ylab("")

temp=milk_WHZ_B %>% arrange(Component_1) %>% mutate(name=paste0(Genus,";",Species)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>% mutate(index=1:nrow(.))
temp[temp$name == "", "name"] = "Unclassified"
e=temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Phylum))) + geom_bar(stat="identity", col="black") + ylab("zOTU") + xlab("Component 1") + scale_y_discrete(labels=temp$name) + scale_fill_manual(name="Phylum", labels=c("Actinobecteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) + theme(legend.position="none") + ylab("")

temp = milkMetab_WHZ_B %>% arrange(Component_1) %>% mutate(index=1:nrow(.))
f = temp %>% ggplot(aes(x=Component_1,y=as.factor(index),fill=as.factor(Class))) + geom_bar_pattern(stat="identity", pattern_angle=45, pattern_density=0.05, pattern_spacing=0.02, pattern_key_scale_factor=0.6, col="black", pattern_fill="black") + ylab("Metabolite") + xlab("Component 1") + scale_y_discrete(labels=temp$Metabolite) + scale_fill_manual(name="Metabolite class", labels=c("Amino acids and derivatives", "Energy related", "Fatty acids and derivatives", "Food derived", "Microbially derived", "Oligosaccharides", "Pharmaceuticals", "Sugars"), values=phylum_cols) + theme(legend.position="none") + ylab("")

a
b
c
d
e
f
```

# Table 4: wilcox and cor tests of all models

```{r table 4 cor tests of all models}
faeces_sampleMeta = read.csv("./faeces_sampleMeta.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(faeces_sampleMeta) = c("ID", "RCID", "BMI", "WeightGroup", "Timepoint", "Gestation", "C_section", "AB_infant", "AB_mother", "Secretor", "lewis", "subject")
milk_sampleMeta = read.csv("./milk_sampleMeta.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(milk_sampleMeta) = c("ID", "RCID", "BMI", "WeightGroup", "Timepoint", "Gestation", "C_section", "AB_infant", "AB_mother", "Secretor", "lewis", "subject")
milkMetab_sampleMeta = read.csv("./milkMetab_sampleMeta.csv", sep=" ", header=FALSE) %>% as_tibble()
colnames(milkMetab_sampleMeta) = c("RCID", "BMI", "WeightGroup", "Timepoint", "Gestation", "C_section", "AB_infant", "AB_mother", "Secretor", "lewis", "subject")

# Load new WHZ metadata
faeces_whz = read.csv("../../Data_for_Roel_w_Anthropometrics/Data_for_Roel_w_Anthropometrics/faeces_asv_meta.csv") %>% as_tibble() %>% select(RCID, whz.6m, whz.1y, whz.2y, whz.3y)
faeces_sampleMeta = faeces_sampleMeta %>% left_join(faeces_whz)

milk_whz = read.csv("../../Data_for_Roel_w_Anthropometrics/Data_for_Roel_w_Anthropometrics/milk_asv_meta.csv") %>% as_tibble() %>% select(RCID, whz.6m, whz.1y, whz.2y, whz.3y)
milk_sampleMeta = milk_sampleMeta %>% left_join(milk_whz)

milkMetab_whz = read.csv("../../Data_for_Roel_w_Anthropometrics/Data_for_Roel_w_Anthropometrics/milk_metab_meta.csv", sep=";", dec=",") %>% as_tibble() %>% select(RCID, whz.6m, whz.1y, whz.2y, whz.3y)
milkMetab_sampleMeta = milkMetab_sampleMeta %>% left_join(milkMetab_whz)

# Load new BMIz metadata
faeces_BMIz = read.csv("./20240906_data/faeces_asv_meta.csv") %>% as_tibble() %>% select(RCID, BMI.delta.3m, delta.whz, BMIz.1y, BMIz.2y, BMIz.3y, BMIz.group.1y, BMIz.group.2y, BMIz.group.3y)
faeces_sampleMeta = faeces_sampleMeta %>% left_join(faeces_BMIz)

milk_BMIz = read.csv("./20240906_data/milk_asv_meta.csv") %>% as_tibble() %>% select(RCID, BMI.delta.3m, delta.whz, BMIz.1y, BMIz.2y, BMIz.3y, BMIz.group.1y, BMIz.group.2y, BMIz.group.3y)
milk_sampleMeta = milk_sampleMeta %>% left_join(milk_BMIz)

milkMetab_BMIz = read.csv("./20240906_data/milk_metab_meta.csv", sep=";", dec=",") %>% as_tibble() %>% select(RCID, BMI.delta.3m, delta.whz, BMIz.1y, BMIz.2y, BMIz.3y, BMIz.group.1y, BMIz.group.2y, BMIz.group.3y)
milkMetab_sampleMeta = milkMetab_sampleMeta %>% left_join(milkMetab_BMIz)

# Prepare test information
faecal_meta_final = faeces_BMI_A %>% mutate(subject = as.numeric(subject)) %>% left_join(faeces_sampleMeta) %>% select(-Timepoint,-RCID) %>% distinct(subject, .keep_all=TRUE) %>% select(-Component_1)
milk_meta_final = milk_BMI_A %>% mutate(subject = as.numeric(subject)) %>% left_join(milk_sampleMeta) %>% select(-Timepoint,-RCID) %>% distinct(subject, .keep_all=TRUE) %>% select(-Component_1)
milkMetab_meta_final = milkMetab_BMI_A %>% mutate(subject = as.numeric(subject)) %>% left_join(milkMetab_sampleMeta) %>% select(-Timepoint,-RCID) %>% distinct(subject, .keep_all=TRUE) %>% select(-Component_1)

p_raw = matrix(0L, nrow=6, ncol=10) # rows: models, cols: tests

## Test correlation with BMI
p_raw[1,1] = cor.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(BMI) %>% pull(),
                      faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[2,1] = cor.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(BMI) %>% pull(),
                      faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[3,1] = cor.test(milk_BMI_A %>% left_join(milk_meta_final) %>% select(BMI) %>% pull(),
                      milk_BMI_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[4,1] = cor.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% select(BMI) %>% pull(),
                      milk_WHZ_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[5,1] = cor.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(BMI) %>% pull(),
                      milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[6,1] = cor.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(BMI) %>% pull(),
                      milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value

## Birth mode
p_raw[1,2] = wilcox.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(C_section==1) %>% select(Component_1) %>% pull(),
                         faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(C_section==2) %>% select(Component_1) %>% pull())$p.value
p_raw[2,2] = wilcox.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(C_section==1) %>% select(Component_1) %>% pull(),
                         faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(C_section==2) %>% select(Component_1) %>% pull())$p.value

p_raw[3,2] = wilcox.test(milk_BMI_A %>% left_join(milk_meta_final) %>% filter(C_section==1) %>% select(Component_1) %>% pull(),
                         milk_BMI_A %>% left_join(milk_meta_final) %>% filter(C_section==2) %>% select(Component_1) %>% pull())$p.value
p_raw[4,2] = wilcox.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(C_section==1) %>% select(Component_1) %>% pull(),
                         milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(C_section==2) %>% select(Component_1) %>% pull())$p.value

p_raw[5,2] = wilcox.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(C_section==1) %>% select(Component_1) %>% pull(),
                         milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(C_section==2) %>% select(Component_1) %>% pull())$p.value
p_raw[6,2] = wilcox.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(C_section==1) %>% select(Component_1) %>% pull(),
                         milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(C_section==2) %>% select(Component_1) %>% pull())$p.value

## AB_infant - not significant
p_raw[1,3] = wilcox.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(AB_infant==1) %>% select(Component_1) %>% pull(),
                         faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(AB_infant==2) %>% select(Component_1) %>% pull())$p.value
p_raw[2,3] = wilcox.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(AB_infant==1) %>% select(Component_1) %>% pull(),
                         faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(AB_infant==2) %>% select(Component_1) %>% pull())$p.value

p_raw[3,3] = wilcox.test(milk_BMI_A %>% left_join(milk_meta_final) %>% filter(AB_infant==1) %>% select(Component_1) %>% pull(),
                         milk_BMI_A %>% left_join(milk_meta_final) %>% filter(AB_infant==2) %>% select(Component_1) %>% pull())$p.value
p_raw[4,3] = wilcox.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(AB_infant==1) %>% select(Component_1) %>% pull(),
                         milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(AB_infant==2) %>% select(Component_1) %>% pull())$p.value

p_raw[5,3] = wilcox.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(AB_infant==1) %>% select(Component_1) %>% pull(),
                         milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(AB_infant==2) %>% select(Component_1) %>% pull())$p.value
p_raw[6,3] = wilcox.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(AB_infant==1) %>% select(Component_1) %>% pull(),
                         milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(AB_infant==2) %>% select(Component_1) %>% pull())$p.value

## AB_mother - not significant
p_raw[1,4] = wilcox.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(AB_mother==1) %>% select(Component_1) %>% pull(),
                         faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(AB_mother==2) %>% select(Component_1) %>% pull())$p.value
p_raw[2,4] = wilcox.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(AB_mother==1) %>% select(Component_1) %>% pull(),
                         faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(AB_mother==2) %>% select(Component_1) %>% pull())$p.value

p_raw[3,4] = wilcox.test(milk_BMI_A %>% left_join(milk_meta_final) %>% filter(AB_mother==1) %>% select(Component_1) %>% pull(),
                         milk_BMI_A %>% left_join(milk_meta_final) %>% filter(AB_mother==2) %>% select(Component_1) %>% pull())$p.value
p_raw[4,4] = wilcox.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(AB_mother==1) %>% select(Component_1) %>% pull(),
                         milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(AB_mother==2) %>% select(Component_1) %>% pull())$p.value

p_raw[5,4] = wilcox.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(AB_mother==1) %>% select(Component_1) %>% pull(),
                         milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(AB_mother==2) %>% select(Component_1) %>% pull())$p.value
p_raw[6,4] = wilcox.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(AB_mother==1) %>% select(Component_1) %>% pull(),
                         milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(AB_mother==2) %>% select(Component_1) %>% pull())$p.value

## Secretor - relevant in component 2
p_raw[1,5] = wilcox.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(Secretor==0) %>% select(Component_1) %>% pull(),
                         faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(Secretor==1) %>% select(Component_1) %>% pull())$p.value
p_raw[2,5] = wilcox.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(Secretor==0) %>% select(Component_1) %>% pull(),
                         faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(Secretor==1) %>% select(Component_1) %>% pull())$p.value

p_raw[3,5] = wilcox.test(milk_BMI_A %>% left_join(milk_meta_final) %>% filter(Secretor==0) %>% select(Component_1) %>% pull(),
                         milk_BMI_A %>% left_join(milk_meta_final) %>% filter(Secretor==1) %>% select(Component_1) %>% pull())$p.value
p_raw[4,5] = wilcox.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(Secretor==0) %>% select(Component_1) %>% pull(),
                         milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(Secretor==1) %>% select(Component_1) %>% pull())$p.value

p_raw[5,5] = wilcox.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(Secretor==0) %>% select(Component_1) %>% pull(),
                         milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(Secretor==1) %>% select(Component_1) %>% pull())$p.value
p_raw[6,5] = wilcox.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(Secretor==0) %>% select(Component_1) %>% pull(),
                         milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(Secretor==1) %>% select(Component_1) %>% pull())$p.value

## Lewis - not significant
p_raw[1,6] = wilcox.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(lewis==0) %>% select(Component_1) %>% pull(),
                         faeces_BMI_A %>% left_join(faecal_meta_final) %>% filter(lewis==1) %>% select(Component_1) %>% pull())$p.value
p_raw[2,6] = wilcox.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(lewis==0) %>% select(Component_1) %>% pull(),
                         faeces_WHZ_A %>% left_join(faecal_meta_final) %>% filter(lewis==1) %>% select(Component_1) %>% pull())$p.value

p_raw[3,6] = wilcox.test(milk_BMI_A %>% left_join(milk_meta_final) %>% filter(lewis==0) %>% select(Component_1) %>% pull(),
                         milk_BMI_A %>% left_join(milk_meta_final) %>% filter(lewis==1) %>% select(Component_1) %>% pull())$p.value
p_raw[4,6] = wilcox.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(lewis==0) %>% select(Component_1) %>% pull(),
                         milk_WHZ_A %>% left_join(milk_meta_final) %>% filter(lewis==1) %>% select(Component_1) %>% pull())$p.value

p_raw[5,6] = wilcox.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(lewis==0) %>% select(Component_1) %>% pull(),
                         milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% filter(lewis==1) %>% select(Component_1) %>% pull())$p.value
p_raw[6,6] = wilcox.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(lewis==0) %>% select(Component_1) %>% pull(),
                         milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% filter(lewis==1) %>% select(Component_1) %>% pull())$p.value

## WHZ 6m
p_raw[1,7] = cor.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(whz.6m) %>% pull(),
                       faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[2,7] = cor.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(whz.6m) %>% pull(),
                       faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[3,7] = cor.test(milk_BMI_A %>% left_join(milk_meta_final) %>% select(whz.6m) %>% pull(),
                       milk_BMI_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[4,7] = cor.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% select(whz.6m) %>% pull(),
                       milk_WHZ_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[5,7] = cor.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(whz.6m) %>% pull(),
                       milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[6,7] = cor.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(whz.6m) %>% pull(),
                       milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value

## WHZ 1y
p_raw[1,8] = cor.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(whz.1y) %>% pull(),
                       faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[2,8] = cor.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(whz.1y) %>% pull(),
                       faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[3,8] = cor.test(milk_BMI_A %>% left_join(milk_meta_final) %>% select(whz.1y) %>% pull(),
                       milk_BMI_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[4,8] = cor.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% select(whz.1y) %>% pull(),
                       milk_WHZ_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[5,8] = cor.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(whz.1y) %>% pull(),
                       milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[6,8] = cor.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(whz.1y) %>% pull(),
                       milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value

## BMIz 2y
p_raw[1,9] = cor.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(BMIz.2y) %>% pull(),
                       faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[2,9] = cor.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(BMIz.2y) %>% pull(),
                       faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[3,9] = cor.test(milk_BMI_A %>% left_join(milk_meta_final) %>% select(BMIz.2y) %>% pull(),
                       milk_BMI_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[4,9] = cor.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% select(BMIz.2y) %>% pull(),
                       milk_WHZ_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[5,9] = cor.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(BMIz.2y) %>% pull(),
                       milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[6,9] = cor.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(BMIz.2y) %>% pull(),
                       milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value

## BMIz 3y
p_raw[1,10] = cor.test(faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(BMIz.3y) %>% pull(),
                       faeces_BMI_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[2,10] = cor.test(faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(BMIz.3y) %>% pull(),
                       faeces_WHZ_A %>% left_join(faecal_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[3,10] = cor.test(milk_BMI_A %>% left_join(milk_meta_final) %>% select(BMIz.3y) %>% pull(),
                       milk_BMI_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[4,10] = cor.test(milk_WHZ_A %>% left_join(milk_meta_final) %>% select(BMIz.3y) %>% pull(),
                      milk_WHZ_A %>% left_join(milk_meta_final) %>% select(Component_1) %>% pull())$p.value

p_raw[5,10] = cor.test(milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(BMIz.3y) %>% pull(),
                       milkMetab_BMI_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value
p_raw[6,10] = cor.test(milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(BMIz.3y) %>% pull(),
                       milkMetab_WHZ_A %>% left_join(milkMetab_meta_final) %>% select(Component_1) %>% pull())$p.value

# Correct final result
p_corrected = matrix(p.adjust(p_raw, method="BH"), nrow=6, ncol=10)
signif(p_corrected, digits=2)
```

```{r figure S4 showing significant aspects of the models}
comparisons = list(c("NW", "OW"), c("NW", "OB"), c("OW", "OB"))

a = faeces_BMI_A %>% ggplot(aes(x=as.factor(WeightGroup),y=Component_1)) + geom_boxplot() + stat_compare_means(comparisons=comparisons,label="p.signif") + xlab("BMI group") + ylab("Subject loading") + ggtitle("Infant faecal microbiome NPLS BMI")
b = faeces_WHZ_A %>% ggplot(aes(x=as.factor(WeightGroup),y=Component_1)) + geom_boxplot() + stat_compare_means(comparisons=comparisons,label="p.signif") + xlab("BMI group") + ylab("Subject loading") + ggtitle("Infant faecal microbiome NPLS WHZ")

c = milk_BMI_A %>% ggplot(aes(x=as.factor(WeightGroup),y=Component_1)) + geom_boxplot() + stat_compare_means(comparisons=comparisons,label="p.signif") + xlab("BMI group") + ylab("Subject loading") + ggtitle("Mother milk microbiome NPLS BMI")
d = milk_WHZ_A %>% ggplot(aes(x=as.factor(WeightGroup),y=Component_1)) + geom_boxplot() + stat_compare_means(comparisons=comparisons,label="p.signif") + xlab("BMI group") + ylab("Subject loading") + ggtitle("Mother milk microbiome NPLS WHZ")

e = milkMetab_BMI_A %>% ggplot(aes(x=as.factor(WeightGroup),y=Component_1)) + geom_boxplot() + stat_compare_means(comparisons=comparisons,label="p.signif") + xlab("BMI group") + ylab("Subject loading") + ggtitle("Mother milk metabolome NPLS BMI")
f = milkMetab_WHZ_A %>% ggplot(aes(x=as.factor(WeightGroup),y=Component_1)) + geom_boxplot() + stat_compare_means(comparisons=comparisons,label="p.signif") + xlab("BMI group") + ylab("Subject loading") + ggtitle("Mother milk metabolome NPLS WHZ")

ggarrange(a,b,c,d,e,f, nrow=3, ncol=2)

```

# NPLS predictive power
```{r Table Sx predictive value of the NPLS models}
a=read.csv("./20240110_faeces_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% ggplot(aes(x=V1,y=V4)) + geom_point() + ggtitle("Infant faecal microbiome NPLS BMI") + stat_cor() + xlab("Predicted BMI") + ylab("Real BMI")
b=read.csv("./20240110_faeces_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% ggplot(aes(x=V1,y=V4)) + geom_point() + ggtitle("Infant faecal microbiome NPLS WHZ") + stat_cor() + xlab("Predicted WHZ") + ylab("Real WHZ")

c=read.csv("./20240110_milk_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% ggplot(aes(x=V1,y=V4)) + geom_point() + ggtitle("Mother milk microbiome NPLS BMI") + stat_cor() + xlab("Predicted BMI") + ylab("Real BMI")
d=read.csv("./20240110_milk_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% ggplot(aes(x=V1,y=V4)) + geom_point() + ggtitle("Mother milk microbiome NPLS WHZ") + stat_cor() + xlab("Predicted WHZ") + ylab("Real WHZ")

e=read.csv("./20240110_milkMetab_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% ggplot(aes(x=V1,y=V4)) + geom_point() + ggtitle("Mother milk metabolomics NPLS BMI") + stat_cor() + xlab("Predicted BMI") + ylab("Real BMI")
f=read.csv("./20240110_milkMetab_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% ggplot(aes(x=V1,y=V4)) + geom_point() + ggtitle("Mother milk metabolomics NPLS WHZ") + stat_cor() + xlab("Predicted WHZ") + ylab("Real WHZ")

ggarrange(a,b,c,d,e,f, nrow=3, ncol=2)

# RMSE
temp = read.csv("./20240110_faeces_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% mutate(missPred = V4 - V1) %>% select(missPred) %>% pull()
sqrt(sum(temp^2) / length(temp))

temp = read.csv("./20240110_milk_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% mutate(missPred = V4 - V1) %>% select(missPred) %>% pull()
sqrt(sum(temp^2) / length(temp))

temp = read.csv("./20240110_milkMetab_NPLS_BMI/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% mutate(missPred = V4 - V1) %>% select(missPred) %>% pull()
sqrt(sum(temp^2) / length(temp))

temp = read.csv("./20240110_faeces_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% mutate(missPred = V4 - V1) %>% select(missPred) %>% pull()
sqrt(sum(temp^2) / length(temp))

temp = read.csv("./20240110_milk_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% mutate(missPred = V4 - V1) %>% select(missPred) %>% pull()
sqrt(sum(temp^2) / length(temp))

temp = read.csv("./20240110_milkMetab_NPLS_WHZ_6m/Rasmus_1_ypred.csv", header=FALSE) %>% as_tibble() %>% mutate(missPred = V4 - V1) %>% select(missPred) %>% pull()
sqrt(sum(temp^2) / length(temp))
```

# Comparison of loadings in faecal and milk microbiomes
```{r microbiomes comparison}
a=faeces_BMI_B %>% left_join(milk_BMI_B, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Phylum.x),label=name)) + geom_point() + xlab("Feature loading in faecal microbiome") + ylab("Feature loading in milk microbiome") + ggtitle("NPLS BMI comparison") + geom_text_repel()

b=faeces_WHZ_B %>% left_join(milk_WHZ_B, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Phylum.x),label=name)) + geom_point() + xlab("Feature loading in faecal microbiome") + ylab("Feature loading in milk microbiome") + ggtitle("NPLS WHZ comparison") + geom_text_repel()

c=faeces_BMI_B %>% left_join(faeces_WHZ_B, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Phylum.x),label=name)) + geom_point() + xlab("Feature loading in BMI model") + ylab("Feature loading in WHZ model") + ggtitle("Faecal models") + geom_text_repel()

d=milk_BMI_B %>% left_join(milk_WHZ_B, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Phylum.x),label=name)) + geom_point() + xlab("Feature loading in BMI model") + ylab("Feature loading in WHZ model") + ggtitle("Milk models") + geom_text_repel()

ggarrange(a,b,c,d)
```

```{r venn approach}
# library(VennDiagram)
# 
# # Faecal
# set1 = faeces_BMI_B %>% filter(Component_1 > 0) %>% select(OTU) %>% pull()
# set2 = faeces_WHZ_B %>% filter(Component_1 > 0) %>% select(OTU) %>% pull()
# set3 = faeces_BMI_B %>% filter(Component_1 < 0) %>% select(OTU) %>% pull()
# set4 = faeces_WHZ_B %>% filter(Component_1 < 0) %>% select(OTU) %>% pull()
# VennDiagram::venn.diagram(x = list(set1, set2,set3,set4), category.names=c("High ppBMI", "High WHZ","Low ppBMI","Low WHZ"), fill=hue_pal()(4), filename="Venn_faecal.png")
# 
# # Milk microbiome
# set1 = milk_BMI_B %>% filter(Component_1 > 0) %>% select(OTU) %>% pull()
# set2 = milk_WHZ_B %>% filter(Component_1 > 0) %>% select(OTU) %>% pull()
# set3 = milk_BMI_B %>% filter(Component_1 < 0) %>% select(OTU) %>% pull()
# set4 = milk_WHZ_B %>% filter(Component_1 < 0) %>% select(OTU) %>% pull()
# VennDiagram::venn.diagram(x = list(set1, set2,set3,set4), category.names=c("High ppBMI", "High WHZ","Low ppBMI","Low WHZ"), fill=hue_pal()(4), filename="Venn_milk.png")
# 
# # Milk metabolome
# set1 = milkMetab_BMI_B %>% filter(Component_1 > 0) %>% select(Metabolite) %>% pull()
# set2 = milkMetab_WHZ_B %>% filter(Component_1 > 0) %>% select(Metabolite) %>% pull()
# set3 = milkMetab_BMI_B %>% filter(Component_1 < 0) %>% select(Metabolite) %>% pull()
# set4 = milkMetab_WHZ_B %>% filter(Component_1 < 0) %>% select(Metabolite) %>% pull()
# VennDiagram::venn.diagram(x = list(set1, set2,set3,set4), category.names=c("High ppBMI", "High WHZ","Low ppBMI","Low WHZ"), fill=hue_pal()(4), filename="Venn_milkMetab.png")
```

```{r overlap manual}

# Faecal
faeces_BMI_B %>% inner_join(faeces_WHZ_B, by="OTU") %>% filter(Component_1.x > 0, Component_1.y > 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))
faeces_BMI_B %>% inner_join(faeces_WHZ_B, by="OTU") %>% filter(Component_1.x < 0, Component_1.y < 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))
faeces_BMI_B %>% inner_join(faeces_WHZ_B, by="OTU") %>% filter(Component_1.x > 0, Component_1.y < 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))
faeces_BMI_B %>% inner_join(faeces_WHZ_B, by="OTU") %>% filter(Component_1.x < 0, Component_1.y > 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))

# Milk
milk_BMI_B %>% inner_join(milk_WHZ_B, by="OTU") %>% filter(Component_1.x > 0, Component_1.y > 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))
milk_BMI_B %>% inner_join(milk_WHZ_B, by="OTU") %>% filter(Component_1.x < 0, Component_1.y < 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))
milk_BMI_B %>% inner_join(milk_WHZ_B, by="OTU") %>% filter(Component_1.x > 0, Component_1.y < 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))
milk_BMI_B %>% inner_join(milk_WHZ_B, by="OTU") %>% filter(Component_1.x < 0, Component_1.y > 0) %>% select(Genus.x, Species.x) %>% arrange(Genus.x, Species.x) %>% print(n=nrow(.))

# Milk metab
milkMetab_BMI_B %>% inner_join(milkMetab_WHZ_B, by="Metabolite") %>% filter(Component_1.x > 0, Component_1.y > 0) %>% select(Metabolite) %>% arrange(Metabolite) %>% print(n=nrow(.))
milkMetab_BMI_B %>% inner_join(milkMetab_WHZ_B, by="Metabolite") %>% filter(Component_1.x < 0, Component_1.y < 0) %>% select(Metabolite) %>% arrange(Metabolite) %>% print(n=nrow(.))
milkMetab_BMI_B %>% inner_join(milkMetab_WHZ_B, by="Metabolite") %>% filter(Component_1.x > 0, Component_1.y < 0) %>% select(Metabolite) %>% arrange(Metabolite) %>% print(n=nrow(.))
milkMetab_BMI_B %>% inner_join(milkMetab_WHZ_B, by="Metabolite") %>% filter(Component_1.x < 0, Component_1.y > 0) %>% select(Metabolite) %>% arrange(Metabolite) %>% print(n=nrow(.))

```

## Rasmus - loadings plots

```{r calculate mean abundances}
faeces_relAbs = sweep(faeces_raw, 1, rowSums(faeces_raw), FUN="/") %>% as_tibble()
milk_relAbs = sweep(milk_raw, 1, rowSums(milk_raw), FUN="/") %>% as_tibble()

faeces_BMI_B_mean = apply(faeces_relAbs, 2, function(x){mean(x, na.rm=TRUE)}) %>% as_tibble() %>% mutate(OTU = faeces_taxa$OTU) %>% right_join(faeces_BMI_B, by="OTU")
faeces_WHZ_B_mean = apply(faeces_relAbs, 2, function(x){mean(x, na.rm=TRUE)}) %>% as_tibble() %>% mutate(OTU = faeces_taxa$OTU) %>% right_join(faeces_WHZ_B, by="OTU")

milk_BMI_B_mean = apply(milk_relAbs, 2, function(x){mean(x, na.rm=TRUE)}) %>% as_tibble() %>% mutate(OTU = milk_taxa$OTU) %>% right_join(milk_BMI_B, by="OTU")
milk_WHZ_B_mean = apply(milk_relAbs, 2, function(x){mean(x, na.rm=TRUE)}) %>% as_tibble() %>% mutate(OTU = milk_taxa$OTU) %>% right_join(milk_WHZ_B, by="OTU")
```

### Non-filtered

```{r Loadings comparison plots showing mean abundances}
#Define phylum colors
colours_phy <- c(
  "Firmicutes" = colours[1], 
  "Bacteroidota" = colours[2], 
  "Actinobacteriota" = colours[3], 
  "Proteobacteria" = colours[4], 
  "Verrucomicrobiota" = colours[5]
)

a=faeces_BMI_B_mean %>% left_join(milk_BMI_B_mean, by="OTU")  %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in faecal microbiome") +
  ylab("Feature loading in milk microbiome") +
  ggtitle("NPLS BMI comparison") +
  geom_text_repel(size = 3) +
  theme_classic() +
  geom_point(aes(size = value.x, shape = "Infant faeces"), fill = "red", alpha = 1, show.legend = TRUE) +
  geom_point(aes(size = value.y, shape = "Milk"), alpha = 1, position = position_nudge(x = 0.015, y = 0), show.legend = TRUE) +
  scale_size_continuous(name = "Relative abundance", guide = guide_legend(order = 1)) +
  scale_shape_manual(name = "Sample type", values = c("Infant faeces" = 16, "Milk" = 1), guide = guide_legend(override.aes = list(size = 5))) +
  #guides(col = guide_legend(override.aes = list(size = 4, shape = 16)),nrow=2) +
  guides(col = "none", shape = guide_legend(override.aes = list(size = 5)), size = guide_legend(order = 1)) +
  scale_color_manual(values = colours_phy,name="Phylum")

a

b=faeces_WHZ_B_mean %>% left_join(milk_WHZ_B_mean, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in faecal microbiome") +
  ylab("Feature loading in milk microbiome") +
  ggtitle("NPLS WHZ comparison") +
  geom_text_repel(size = 3) +
  theme_classic() +
  geom_point(aes(size = value.x, shape = "Infant faeces"), fill = "red", alpha = 1, show.legend = TRUE) +
  geom_point(aes(size = value.y, shape = "Milk"), alpha = 1, position = position_nudge(x = 0.015, y = 0), show.legend = TRUE) +
  scale_size_continuous(name = "Relative abundance", guide = guide_legend(order = 1)) +
  scale_shape_manual(name = "Sample type", values = c("Infant faeces" = 16, "Milk" = 1), guide = guide_legend(override.aes = list(size = 5))) +
  guides(col = guide_legend(override.aes = list(size = 4, shape = 16)),nrow=2)+
  scale_color_manual(values = colours_phy,name="Phylum")

c=faeces_BMI_B_mean %>% left_join(faeces_WHZ_B_mean, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in BMI model") +
  ylab("Feature loading in WHZ model") +
  ggtitle("NPLS infant faeces BMI-WHZ comparison") +
  geom_text_repel(size = 3) +
  theme_classic() +
  geom_point(aes(size = value.x), alpha = 1, show.legend = TRUE, )+
  scale_size_continuous(name = "Relative abundance")+
  scale_color_manual(values = colours_phy,name="Phylum")

d=milk_BMI_B_mean %>% left_join(milk_WHZ_B_mean, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in BMI model") +
  ylab("Feature loading in WHZ model") +
  ggtitle("NPLS milk BMI-WHZ comparison") +
  geom_text_repel(size = 3) +
  theme_classic() +
  geom_point(aes(size = value.x), alpha = 1, shape=1, show.legend = TRUE, )+
  scale_size_continuous(name = "Relative abundance")+
  scale_color_manual(values = colours_phy,name="Phylum")

ggarrange(a+theme(legend.position = "right"),
          b+theme(legend.position = "right"),
          c+theme(legend.position = "right"),
          d+theme(legend.position = "right"))
```

### Filtered

```{r Loadings comparison plots filtered by variance explained}
Xhat_faeces_BMI = k_unfold(faeces_BMI_modelled, 2)@data
Xhat_faeces_WHZ = k_unfold(faeces_WHZ_modelled, 2)@data
Xhat_milk_BMI = k_unfold(milk_BMI_modelled, 2)@data
Xhat_milk_WHZ = k_unfold(milk_WHZ_modelled, 2)@data

X_faeces_BMI = k_unfold(faeces_BMI_input, 2)@data
X_faeces_WHZ = k_unfold(faeces_WHZ_input, 2)@data
X_milk_BMI = k_unfold(milk_BMI_input, 2)@data
X_milk_WHZ = k_unfold(milk_WHZ_input, 2)@data

faeces_BMI_B_mean$varexp = (rowSums(Xhat_faeces_BMI^2, na.rm=TRUE) / rowSums(X_faeces_BMI^2, na.rm=TRUE))
faeces_WHZ_B_mean$varexp  = (rowSums(Xhat_faeces_WHZ^2, na.rm=TRUE) / rowSums(X_faeces_WHZ^2, na.rm=TRUE))

milk_BMI_B_mean$varexp = (rowSums(Xhat_milk_BMI^2, na.rm=TRUE) / rowSums(X_milk_BMI^2, na.rm=TRUE))
milk_WHZ_B_mean$varexp = (rowSums(Xhat_milk_WHZ^2, na.rm=TRUE) / rowSums(X_milk_WHZ^2, na.rm=TRUE))

#Set quantile threshold
quant <- 0.4
quant.cd <- 0.85

a=faeces_BMI_B_mean %>% left_join(milk_BMI_B_mean, by="OTU")  %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
  mutate(quant_x = quantile(abs(varexp.x), quant, na.rm = TRUE),
         quant_y = quantile(abs(varexp.y), quant, na.rm = TRUE)) %>%
  filter(!(abs(varexp.x) < quant_x & abs(varexp.x) < quant_y)) %>%
  select(-quant_x, -quant_y) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in faecal microbiome") +
  ylab("Feature loading in milk microbiome") +
  ggtitle("NPLS BMI comparison") +
  scale_color_manual(values = colours_phy,name="Phylum") +
  geom_text_repel(size = 3.5,alpha=1) +
  theme_classic() +
  geom_point(aes(size = value.x, shape = "Infant faeces"), fill = "red", alpha = 0.7, show.legend = TRUE) +
  geom_point(aes(size = value.y, shape = "Milk"), alpha = 0.7, position = position_nudge(x = 0.015, y = 0), show.legend = TRUE) +
  scale_size_continuous(name = "Relative abundance", guide = guide_legend(order = 1)) +
  scale_shape_manual(name = "Sample type", values = c("Infant faeces" = 16, "Milk" = 1), guide = guide_legend(override.aes = list(size = 5))) +
  guides(col = guide_legend(override.aes = list(size = 4, shape = 16)))

a

b=faeces_WHZ_B_mean %>% left_join(milk_WHZ_B_mean, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
     mutate(quant_x = quantile(abs(varexp.x), quant, na.rm = TRUE),
         quant_y = quantile(abs(varexp.y), quant, na.rm = TRUE)) %>%
  filter(!(abs(varexp.x) < quant_x & abs(varexp.x) < quant_y)) %>%
  select(-quant_x, -quant_y) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in faecal microbiome") +
  ylab("Feature loading in milk microbiome") +
  ggtitle("NPLS WHZ comparison") +
 scale_color_manual(values = colours_phy,name="Phylum") +
  geom_text_repel(size = 3.5,alpha=1) +
  theme_classic() +
  geom_point(aes(size = value.x, shape = "Infant faeces"), fill = "red", alpha = 0.7, show.legend = TRUE) +
  geom_point(aes(size = value.y, shape = "Milk"), alpha = 0.7, position = position_nudge(x = 0.015, y = 0), show.legend = TRUE) +
  scale_size_continuous(name = "Relative abundance", guide = guide_legend(order = 1)) +
  scale_shape_manual(name = "Sample type", values = c("Infant faeces" = 16, "Milk" = 1), guide = guide_legend(override.aes = list(size = 5))) +
  guides(col = guide_legend(override.aes = list(size = 4, shape = 16)))

c=faeces_BMI_B_mean %>% left_join(faeces_WHZ_B_mean, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
   mutate(quant_x = quantile(abs(varexp.x), quant.cd, na.rm = TRUE),
         quant_y = quantile(abs(varexp.y), quant.cd, na.rm = TRUE)) %>%
  filter(!(abs(varexp.x) < quant_x & abs(varexp.y) < quant_y)) %>%
  select(-quant_x, -quant_y) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in BMI model") +
  ylab("Feature loading in WHZ model") +
  ggtitle("NPLS infant faeces BMI-WHZ comparison") +
  scale_color_manual(values = colours_phy,name="Phylum") +
  geom_text_repel(size = 3.5,alpha=1) +
  theme_classic() +
  geom_point(aes(size = value.x), alpha = 0.7, show.legend = TRUE, )+
  scale_size_continuous(name = "Relative abundance")

d=milk_BMI_B_mean %>% left_join(milk_WHZ_B_mean, by="OTU") %>% mutate(name=paste0(Genus.x,";",Species.x)) %>% mutate(name = gsub("s__uncultured[_\\w]*.*|g__", "", name)) %>% mutate(name = gsub("|.*s__|;", "", name)) %>%  #Split gsub to ensure sequence of operations
  mutate(quant_x = quantile(abs(varexp.x), quant.cd, na.rm = TRUE),
         quant_y = quantile(abs(varexp.y), quant.cd, na.rm = TRUE)) %>%
  filter(!(abs(varexp.x) < quant_x & abs(varexp.y) < quant_y)) %>%
  select(-quant_x, -quant_y) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(gsub("p__","",Phylum.x)),label=name)) +
  xlab("Feature loading in BMI model") +
  ylab("Feature loading in WHZ model") +
  ggtitle("NPLS milk BMI-WHZ comparison") +
  scale_color_manual(values = colours_phy,name="Phylum") +
  geom_text_repel(size = 3.5,alpha=1) +
  theme_classic() +
  geom_point(aes(size = value.x),shape=1, alpha = 0.7, show.legend = TRUE, )+
  scale_size_continuous(name = "Relative abundance")

ggarrange(a,b,c,d)
```
### Roels attempt at figure 4
```{r figure 4 loading comparison}

# Faecal BMI vs WHZ
temp = faeces_BMI_B_mean %>% 
  select(OTU,Component_1) %>% 
  inner_join(faeces_WHZ_B_mean, by="OTU") %>%
  mutate(Genus = gsub("g__", "", Genus), Species = gsub("s__", "", Species)) %>% 
  mutate(Species = str_split_fixed(Species, "_", 2)[,2]) %>% 
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "sp."), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "organism"), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "bacterium"), "", .x)))

temp[temp$Species == "", "Species"] = "sp."
temp$name = paste0(temp$Genus, " ", temp$Species)
temp[temp$name == " sp.", "name"] = "Unclassified"

a = temp %>% 
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Phylum),label=name)) +
  theme_classic() +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_point(aes(size=value)) +
  geom_text_repel(fontface="italic", max.overlaps=6) +
  xlab("Feature loading in ppBMI model") +
  ylab("Feature loading in infant 6-month WHZ model") +
  scale_colour_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols[-3]) +
  theme(legend.position="none")

# HM microbiome BMI vs WHZ
temp = milk_BMI_B_mean %>% 
  select(OTU,Component_1) %>% 
  inner_join(milk_WHZ_B_mean, by="OTU") %>%
  mutate(Genus = gsub("g__", "", Genus), Species = gsub("s__", "", Species)) %>% 
  mutate(Species = str_split_fixed(Species, "_", 2)[,2]) %>% 
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "sp."), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "organism"), "", .x))) %>%
  mutate(dplyr::across(.cols = Species, .fns = ~ dplyr::if_else(stringr::str_detect(.x, "bacterium"), "", .x)))

temp[temp$Species == "", "Species"] = "sp."
temp$name = paste0(temp$Genus, " ", temp$Species)
temp[temp$name == " sp.", "name"] = "Unclassified"

b = temp %>% 
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Phylum),label=name)) +
  theme_classic() +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_point(aes(size=value)) +
  geom_text_repel(fontface="italic", max.overlaps=6) +
  xlab("Feature loading in ppBMI model") +
  ylab("Feature loading in infant 6-month WHZ model") +
  scale_colour_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols) +
  theme(legend.position="none")

# HM metabolome BMI vs WHZ
c = milkMetab_BMI_B %>%
  select(Component_1,Metabolite) %>%
  inner_join(milkMetab_WHZ_B, by="Metabolite") %>% 
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Class),label=Metabolite)) +
  theme_classic() +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_point(shape=17) +
  geom_text_repel(max.overlaps=5) +
  xlab("Feature loading in ppBMI model") +
  ylab("Feature loading in infant 6-month WHZ model") +
  scale_color_manual(values = colours,name="Metabolite class") +
  theme(legend.position="none")

ggarrange(a,b,c)

### Fake for legend
milk_BMI_B_mean %>% 
  ggplot(aes(x=Component_1,y=Component_1,col=as.factor(Phylum))) +
  theme_classic() +
  geom_point(aes(size=value)) +
  xlab("Feature loading in ppBMI model") +
  ylab("Feature loading in infant 6-month WHZ model") +
  scale_colour_manual(name="Phylum", labels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Verrucomicrobiota"), values=phylum_cols) +
  scale_size_continuous(name = "Relative abundance")
  theme(legend.position="top")

milkMetab_BMI_B %>%
  select(Component_1,Metabolite) %>%
  inner_join(milkMetab_WHZ_B, by="Metabolite") %>% 
  filter(abs(Component_1.x) >= 0.1 | abs(Component_1.y) >= 0.1) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Class),label=Metabolite)) +
  theme_classic() +
  geom_point(shape=17) +
  xlab("Feature loading in ppBMI model") +
  ylab("Feature loading in infant 6-month WHZ model") +
  scale_color_manual(values = colours[-c(5,7)],name="Metabolite class") +
  theme(legend.position="top") +
  guides(col=guide_legend(nrow=3,byrow=TRUE))
```

### Metabolite loadings comparisons

```{r metabolite loadings}

e=milkMetab_BMI_B %>% left_join(milkMetab_WHZ_B, by="Metabolite") %>%
  mutate(Metabolite = gsub("X", "", Metabolite)
         #,Metabolite = gsub("\\.", " ", Metabolite)
         ) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Class.x),label=Metabolite)) +
  xlab("Feature loading in BMI model") +
  ylab("Feature loading in WHZ model") +
  ggtitle("NPLS infant faeces BMI-WHZ comparison") +
  scale_color_manual(values = colours,name="Metabolite class") +
  #geom_text_repel(size = 3,alpha=1) +
  theme_classic() +
  geom_label_repel(label.padding = unit(0.2, "lines"), size = 3, label.size = 0.3,force=2) +
  scale_size_continuous(name = "Relative abundance")

e

ggarrange(
  ggarrange(a+theme(legend.position = "none"),
            b+theme(legend.position = "none"),
            c+theme(legend.position = "none"),
            d+theme(legend.position = "none")
            ),
  e,
  nrow=2,
  heights=c(2,1)
)

ggsave("Figure 2.tiff", compression = "lzw",width = 25, height = 35, units = "cm", dpi = 400) 

ggsave("Figure 2.pdf",width = 25, height = 35, units = "cm", dpi = 400)

ggsave("Figure 2.png",width = 25, height = 35, units = "cm", dpi = 400)  
```

```{r metabolite loadings}

a=milkMetab_BMI_B %>% left_join(milkMetab_WHZ_B, by="Metabolite") %>% 
  #mutate(name=paste0(Genus.x,";",Species.x)) %>%
  mutate(quant_x = quantile(abs(Component_1.x), quant.cd, na.rm = TRUE),
         quant_y = quantile(abs(Component_1.y), quant.cd, na.rm = TRUE)) %>%
  filter(!(abs(Component_1.x) < quant_x & abs(Component_1.y) < quant_y)) %>%
  ggplot(aes(x=Component_1.x,y=Component_1.y,col=as.factor(Class.x),label=Metabolite)) +
  xlab("Feature loading in BMI model") +
  ylab("Feature loading in WHZ model") +
  ggtitle("NPLS mother milk metabolomics BMI-WHZ comparison") +
  scale_color_manual(values = colours[-c(5,7)],name="Metabolite class") +
  #geom_text_repel(size = 3,alpha=1) +
  geom_label_repel(label.padding = unit(0.2, "lines"), size = 3, label.size = 0.3,force=2) +
  theme_classic() +
  #(aes(size = 2), alpha = 0.7, show.legend = TRUE, )+
  scale_size_continuous(name = "Relative abundance")

a
```

# Loading vs relative abundance plots

```{r loading vs relabs}
faeces_relAbs = sweep(faeces_raw, 1, rowSums(faeces_raw), FUN="/") %>% as_tibble()
milk_relAbs = sweep(milk_raw, 1, rowSums(milk_raw), FUN="/") %>% as_tibble()
```
# Clustering

```{r Figures S10-S13 clustering statistics}
set.seed(123)

Xhat_faeces_BMI = k_unfold(faeces_BMI_modelled, 2)@data
Xhat_faeces_WHZ = k_unfold(faeces_WHZ_modelled, 2)@data
Xhat_milk_BMI = k_unfold(milk_BMI_modelled, 2)@data
Xhat_milk_WHZ = k_unfold(milk_WHZ_modelled, 2)@data

X_faeces_BMI = k_unfold(faeces_BMI_input, 2)@data
X_faeces_WHZ = k_unfold(faeces_WHZ_input, 2)@data
X_milk_BMI = k_unfold(milk_BMI_input, 2)@data
X_milk_WHZ = k_unfold(milk_WHZ_input, 2)@data

mask_faeces_BMI = (rowSums(Xhat_faeces_BMI^2, na.rm=TRUE) / rowSums(X_faeces_BMI^2, na.rm=TRUE)) >= (multiway::sumsq(Xhat_faeces_BMI, na.rm=TRUE) / multiway::sumsq(X_faeces_BMI, na.rm=TRUE))
mask_faeces_WHZ = (rowSums(Xhat_faeces_WHZ^2, na.rm=TRUE) / rowSums(X_faeces_WHZ^2, na.rm=TRUE)) >= (multiway::sumsq(Xhat_faeces_WHZ, na.rm=TRUE) / multiway::sumsq(X_faeces_WHZ, na.rm=TRUE))

mask_milk_BMI = (rowSums(Xhat_milk_BMI^2, na.rm=TRUE) / rowSums(X_milk_BMI^2, na.rm=TRUE)) >= (multiway::sumsq(Xhat_milk_BMI, na.rm=TRUE) / multiway::sumsq(X_milk_BMI, na.rm=TRUE))
mask_milk_WHZ = (rowSums(Xhat_milk_WHZ^2, na.rm=TRUE) / rowSums(X_milk_WHZ^2, na.rm=TRUE)) >= (multiway::sumsq(Xhat_milk_WHZ, na.rm=TRUE) / multiway::sumsq(X_milk_WHZ, na.rm=TRUE))

a=fviz_nbclust(Xhat_faeces_BMI[mask_faeces_BMI,], pam, method="wss", k.max=5)
b=fviz_nbclust(Xhat_faeces_BMI[mask_faeces_BMI,], pam, method="silhouette", k.max=5)
c=fviz_nbclust(Xhat_faeces_BMI[mask_faeces_BMI,], pam, method="gap_stat", k.max=5)
ggarrange(a,b,c, nrow=3)

a=fviz_nbclust(Xhat_faeces_WHZ[mask_faeces_WHZ,], pam, method="wss", k.max=5)
b=fviz_nbclust(Xhat_faeces_WHZ[mask_faeces_WHZ,], pam, method="silhouette", k.max=5)
c=fviz_nbclust(Xhat_faeces_WHZ[mask_faeces_WHZ,], pam, method="gap_stat", k.max=5)
ggarrange(a,b,c, nrow=3)

a=fviz_nbclust(Xhat_milk_BMI[mask_milk_BMI,], pam, method="wss", k.max=5)
b=fviz_nbclust(Xhat_milk_BMI[mask_milk_BMI,], pam, method="silhouette", k.max=5)
c=fviz_nbclust(Xhat_milk_BMI[mask_milk_BMI,], pam, method="gap_stat", k.max=5)
ggarrange(a,b,c, nrow=3)

a=fviz_nbclust(Xhat_milk_WHZ[mask_milk_WHZ,], pam, method="wss", k.max=5)
b=fviz_nbclust(Xhat_milk_WHZ[mask_milk_WHZ,], pam, method="silhouette", k.max=5)
c=fviz_nbclust(Xhat_milk_WHZ[mask_milk_WHZ,], pam, method="gap_stat", k.max=5)
ggarrange(a,b,c, nrow=3)
```

```{r actual clustering}
cluster_faeces_BMI = faeces_BMI_B[mask_faeces_BMI,] %>% mutate(cluster = pam(Xhat_faeces_BMI[mask_faeces_BMI,], 2, nstart=50)$cluster)
cluster_faeces_WHZ = faeces_WHZ_B[mask_faeces_WHZ,] %>% mutate(cluster = pam(Xhat_faeces_WHZ[mask_faeces_WHZ,], 2, nstart=50)$cluster)
cluster_milk_BMI = milk_BMI_B[mask_milk_BMI,] %>% mutate(cluster = pam(Xhat_milk_BMI[mask_milk_BMI,], 2, nstart=50)$cluster)
cluster_milk_WHZ = milk_WHZ_B[mask_milk_WHZ,] %>% mutate(cluster = pam(Xhat_milk_WHZ[mask_milk_WHZ,], 2, nstart=50)$cluster)

cluster_faeces_BMI %>% filter(cluster==1) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
cluster_faeces_BMI %>% filter(cluster==2) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
cluster_faeces_WHZ %>% filter(cluster==1) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
cluster_faeces_WHZ %>% filter(cluster==2) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))

cluster_milk_BMI %>% filter(cluster==1) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
cluster_milk_BMI %>% filter(cluster==2) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
cluster_milk_WHZ %>% filter(cluster==1) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
cluster_milk_WHZ %>% filter(cluster==2) %>% select(Component_1,Genus,Species) %>% filter(Species != "", Genus != "") %>% arrange(Genus,Species) %>% print(n=nrow(.))
```

```{r Fig 2 and S9 cluster plotting of NW vs OWOB}

# Infant faecal - BMI
temp = faeces_relAbs %>% select(cluster_faeces_BMI$OTU) %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_faeces_BMI) %>% filter(cluster %in% c(1,2)) %>% group_by(subject,timepoint,cluster) %>% summarize(s = sum(value)) %>% left_join(faecal_meta_final) %>% select(subject,timepoint,cluster,s,WeightGroup) %>% ungroup() %>% filter(WeightGroup %in% c("NW", "OW", "OB")) %>% group_by(WeightGroup, timepoint, cluster) %>% summarize(m = mean(s), v=plotrix::std.error(s)) %>% ungroup() %>% mutate(cluster_num = paste0("Cluster", " ", cluster)) %>% mutate(cluster_name = "Cluster 1: \n High ppBMI associated")
temp[temp$cluster_num == "Cluster 2", "cluster_name"] = "Cluster 2: \n Low ppBMI associated"

a = temp %>% ggplot(aes(x=timepoint,y=m,col=as.factor(WeightGroup))) + facet_wrap(~cluster_name) + geom_line() + geom_point() + geom_errorbar(aes(ymin=m-v,ymax=m+v),width=2) + xlab("Time point [days]") + ylab("Relative abundance sum\n (mean +/- SEM)") + scale_color_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + theme(legend.position="none", text=element_text(size=12))

# Infant faecal WHZ
# temp = faeces_relAbs %>% select(cluster_faeces_WHZ$OTU) %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_faeces_WHZ) %>% filter(cluster %in% c(1,2)) %>% group_by(subject,timepoint,cluster) %>% summarize(s = sum(value)) %>% left_join(faecal_meta_final) %>% select(subject,timepoint,cluster,s,WeightGroup) %>% ungroup() %>% filter(WeightGroup %in% c("NW", "OW", "OB")) %>% group_by(WeightGroup, timepoint, cluster) %>% summarize(m = mean(s), v=plotrix::std.error(s)) %>% ungroup() %>% mutate(cluster_num = paste0("Cluster", " ", cluster)) %>% mutate(cluster_name = "Cluster 1: \n Low WHZ associated")
# temp[temp$cluster_num == "Cluster 2", "cluster_name"] = "Cluster 2: \n High WHZ associated"
temp = faeces_relAbs %>% select(cluster_faeces_WHZ$OTU) %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_faeces_WHZ) %>% filter(cluster %in% c(1,2)) %>% group_by(subject,timepoint,cluster) %>% summarize(s = sum(value)) %>% left_join(faecal_meta_final) %>% left_join(whz_groups) %>% select(subject,timepoint,cluster,s,whz_group) %>% ungroup() %>% filter(whz_group %in% c("Normal", "Overweight", "Underweight")) %>% group_by(whz_group,timepoint,cluster) %>% summarize(m=mean(s), v=plotrix::std.error(s)) %>% ungroup() %>% mutate(cluster_num = paste0("Cluster", " ", cluster)) %>% mutate(cluster_name = "Cluster 1: \n Low WHZ associated")
temp[temp$cluster_num == "Cluster 2", "cluster_name"] = "Cluster 2: \n High WHZ associated"

# b = temp %>% ggplot(aes(x=timepoint,y=m,col=as.factor(WeightGroup))) + facet_wrap(~cluster_name) + geom_line() + geom_point() + geom_errorbar(aes(ymin=m-v,ymax=m+v), width=2) + xlab("Time point [days]") + ylab("Relative abundance sum\n (mean +/- SEM)") + scale_color_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + theme(legend.position="none", text=element_text(size=12))

b = temp %>% ggplot(aes(x=timepoint,y=m,col=as.factor(whz_group))) + facet_wrap(~cluster_name) + geom_line(linetype="dashed") + geom_point() + geom_errorbar(aes(ymin=m-v,ymax=m+v), width=2) + xlab("Time point [days]") + ylab("Relative abundance sum\n (mean +/- SEM)") + scale_color_manual(name="WHZ group", labels=c("Normal", "Overweight", "Underweight"), values=WHZ_cols) + theme(legend.position="none", text=element_text(size=12))

# Milk BMI

temp = milk_relAbs %>% select(cluster_milk_BMI$OTU) %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_milk_BMI) %>% filter(cluster %in% c(1,2)) %>% group_by(subject,timepoint,cluster) %>% summarize(s = sum(value)) %>% left_join(faecal_meta_final) %>% select(subject,timepoint,cluster,s,WeightGroup) %>% ungroup() %>% filter(WeightGroup %in% c("NW", "OW", "OB")) %>% group_by(WeightGroup, timepoint, cluster) %>% summarize(m = mean(s), v=plotrix::std.error(s)) %>% ungroup() %>% filter(WeightGroup != "NA") %>% mutate(cluster_num = paste0("Cluster", " ", cluster)) %>% mutate(cluster_name = "Cluster 1 \n Low ppBMI associated")
temp[temp$cluster_num == "Cluster 2", "cluster_name"] = "Cluster 2: \n High ppBMI associated"

c = temp %>% ggplot(aes(x=timepoint,y=m,col=as.factor(WeightGroup))) + facet_wrap(~cluster_name) + geom_line() + geom_point() + geom_errorbar(aes(ymin=m-v,ymax=m+v), width=2) + xlab("Time point [days]") + ylab("Relative abundance sum\n (mean +/- SEM)") + scale_color_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + theme(legend.position="none", text=element_text(size=12))

# Milk WHZ
# temp = milk_relAbs %>% select(cluster_milk_WHZ$OTU) %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_milk_WHZ) %>% filter(cluster %in% c(1,2)) %>% group_by(subject,timepoint,cluster) %>% summarize(s = sum(value)) %>% left_join(faecal_meta_final) %>% select(subject,timepoint,cluster,s,WeightGroup) %>% ungroup() %>% filter(WeightGroup %in% c("NW", "OW", "OB")) %>% group_by(WeightGroup, timepoint, cluster) %>% summarize(m = mean(s), v=plotrix::std.error(s)) %>% ungroup() %>% filter(WeightGroup != "NA") %>% mutate(cluster_num = paste0("Cluster", " ", cluster)) %>% mutate(cluster_name = "Cluster 1: \n Low WHZ associated")
# temp[temp$cluster_num == "Cluster 2", "cluster_name"] = "Cluster 2: \n High WHZ associated"

temp = milk_relAbs %>% select(cluster_milk_WHZ$OTU) %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_milk_WHZ) %>% filter(cluster %in% c(1,2)) %>% group_by(subject,timepoint,cluster) %>% summarize(s = sum(value)) %>% left_join(faecal_meta_final) %>% left_join(whz_groups) %>% select(subject,timepoint,cluster,s,whz_group) %>% ungroup() %>% filter(whz_group %in% c("Normal", "Overweight", "Underweight")) %>% group_by(whz_group, timepoint, cluster) %>% summarize(m = mean(s), v=plotrix::std.error(s)) %>% ungroup() %>% mutate(cluster_num = paste0("Cluster", " ", cluster)) %>% mutate(cluster_name = "Cluster 1: \n Low WHZ associated")
temp[temp$cluster_num == "Cluster 2", "cluster_name"] = "Cluster 2: \n High WHZ associated"

# d = temp %>% ggplot(aes(x=timepoint,y=m,col=as.factor(WeightGroup))) + facet_wrap(~cluster_name) + geom_line() + geom_point() + geom_errorbar(aes(ymin=m-v,ymax=m+v), width=2) + xlab("Time point [days]") + ylab("Relative abundance sum\n (mean +/- SEM)") + scale_color_manual(name="BMI group", labels=c("Normal", "Overweight", "Obese"), values=BMI_cols) + theme(legend.position="none", text=element_text(size=12))

d = temp %>% ggplot(aes(x=timepoint,y=m,col=as.factor(whz_group))) + facet_wrap(~cluster_name) + geom_line(linetype="dashed") + geom_point() + geom_errorbar(aes(ymin=m-v,ymax=m+v), width=2) + xlab("Time point [days]") + ylab("Relative abundance sum\n (mean +/- SEM)") + scale_color_manual(name="WHZ group", labels=c("Normal", "Overweight", "Underweight"), values=WHZ_cols) + theme(legend.position="none", text=element_text(size=12))

ggarrange(a,c) # BMI models, fig 2 - left: infant faecal bmi model, right: mother milk microbiome bmi model
ggarrange(b,d) # WHZ models, fig S9

ggarrange(a,c,b,d, nrow=2, ncol=2)
```

```{r Fig 2 and Table Sx permutation testing of NW vs OWOB clusters in the BMI NPLS models}
ASVgroupPermutationTest = function(relativeAbundances, featureClustering, day, clusterNumber, numPermutations){

  df = relativeAbundances %>%
    select(all_of(c(featureClustering$OTU, "subject", "timepoint"))) %>%
    pivot_longer(-c(subject, timepoint)) %>%
    left_join(featureClustering, by=c("name"="OTU")) %>%
    group_by(subject, timepoint, cluster) %>%
    summarize(s=sum(value), .groups="drop") %>%
    ungroup() %>%
    left_join(faeces_sampleMeta %>% select(subject, WeightGroup) %>% unique(), by="subject") %>%
    filter(cluster == clusterNumber, timepoint == day)

  dfA = df %>% filter(WeightGroup=="NW") %>% select(s) %>% pull()
  dfB = df %>% filter(WeightGroup %in% c("OW", "OB")) %>% select(s) %>% pull()
  #realResult = wilcox.test(dfA, dfB)$p.value
  realResult = mean(dfA) - mean(dfB)

  # Permutations
  set.seed(1)
  permutedResults = 1:numPermutations
  for(i in 1:numPermutations){
    dfA = df %>% mutate(WeightGroup = sample(WeightGroup)) %>% filter(WeightGroup=="NW") %>% select(s) %>% pull()
    dfB = df %>% mutate(WeightGroup = sample(WeightGroup)) %>% filter(WeightGroup %in% c("OW", "OB")) %>% select(s) %>% pull()
    #permutedResults[i] = wilcox.test(dfA, dfB)$p.value
    permutedResults[i] = mean(dfA) - mean(dfB)
  }

  Zscore = abs(realResult - mean(permutedResults)) / sd(permutedResults)

  if (realResult < 0){
    pvalue = sum(permutedResults < realResult) / numPermutations
  }
  else{
    pvalue = sum(permutedResults > realResult) / numPermutations
  }

  return(list(realResult, permutedResults, mean(permutedResults), median(permutedResults), sd(permutedResults), Zscore, pvalue))
}

# Faecal samples
faeces_featureClustering = faeces_relAbs %>% select(cluster_faeces_BMI$OTU) %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_faeces_BMI) %>% filter(cluster %in% c(1,2)) %>% select(OTU, cluster) %>% unique()

faeces_relAbs = faeces_relAbs %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint)

uncorrectedP = matrix(NA, nrow=2, ncol=3)
days = c(30, 60, 90)

for(i in 1:length(days)){
  day = days[i]
  for(cluster in 1:2){
    uncorrectedP[cluster, i] = ASVgroupPermutationTest(faeces_relAbs, faeces_featureClustering, day, cluster, 999)[[7]]
  }
}

correctedP_faeces = matrix(p.adjust(uncorrectedP, "BH"), nrow=2, ncol=3)

# Milk samples
milk_featureClustering = milk_relAbs %>% select(cluster_milk_BMI$OTU) %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_milk_BMI) %>% filter(cluster %in% c(1,2)) %>% select(OTU, cluster) %>% unique()

milk_relAbs = milk_relAbs %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint)

uncorrectedP = matrix(NA, nrow=2, ncol=4)
days = c(3, 30, 60, 90)

for(i in 1:length(days)){
  day = days[i]
  for(cluster in 1:2){
    uncorrectedP[cluster, i] = ASVgroupPermutationTest(milk_relAbs, milk_featureClustering, day, cluster, 999)[[7]]
  }
}

correctedP_milk = matrix(p.adjust(uncorrectedP, "BH"), nrow=2, ncol=4)
```

```{r Fig S14 and Table Sx permutation testing of NW vs OWOB clusters in the WHZ NPLS models}
ASVgroupPermutationTest_WHZ = function(relativeAbundances, featureClustering, day, clusterNumber, numPermutations){

  df = relativeAbundances %>%
    select(all_of(c(featureClustering$OTU, "subject", "timepoint"))) %>%
    pivot_longer(-c(subject, timepoint)) %>%
    left_join(featureClustering, by=c("name"="OTU")) %>%
    group_by(subject, timepoint, cluster) %>%
    summarize(s=sum(value), .groups="drop") %>%
    ungroup() %>%
    left_join(faeces_sampleMeta %>% select(subject, WeightGroup) %>% unique(), by="subject") %>%
    filter(cluster == clusterNumber, timepoint == day) %>%
    left_join(whz_groups)

  dfA = df %>% filter(whz_group %in% c("Overweight")) %>% select(s) %>% pull()
  dfB = df %>% filter(whz_group %in% c("Underweight")) %>% select(s) %>% pull()
  #realResult = wilcox.test(dfA, dfB)$p.value
  realResult = mean(dfA) - mean(dfB)

  # Permutations
  set.seed(1)
  permutedResults = 1:numPermutations
  for(i in 1:numPermutations){
    dfA = df %>% mutate(whz_group = sample(whz_group)) %>% filter(whz_group %in% c("Overweight")) %>% select(s) %>% pull()
    dfB = df %>% mutate(whz_group = sample(whz_group)) %>% filter(whz_group %in% c("Underweight")) %>% select(s) %>% pull()
    #permutedResults[i] = wilcox.test(dfA, dfB)$p.value
    permutedResults[i] = mean(dfA) - mean(dfB)
  }

  Zscore = abs(realResult - mean(permutedResults)) / sd(permutedResults)

  if (realResult < 0){
    pvalue = sum(permutedResults < realResult) / numPermutations
  }
  else{
    pvalue = sum(permutedResults > realResult) / numPermutations
  }

  return(list(realResult, permutedResults, mean(permutedResults), median(permutedResults), sd(permutedResults), Zscore, pvalue))
}

# Faecal samples
faeces_featureClustering = faeces_relAbs %>% select(cluster_faeces_WHZ$OTU) %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_faeces_BMI) %>% filter(cluster %in% c(1,2)) %>% select(OTU, cluster) %>% unique()

faeces_relAbs = faeces_relAbs %>% mutate(subject = faeces_sampleMeta$subject, timepoint = faeces_sampleMeta$Timepoint)

uncorrectedP = matrix(NA, nrow=2, ncol=3)
days = c(30, 60, 90)

for(i in 1:length(days)){
  day = days[i]
  for(cluster in 1:2){
    uncorrectedP[cluster, i] = ASVgroupPermutationTest_WHZ(faeces_relAbs, faeces_featureClustering, day, cluster, 999)[[7]]
  }
}

correctedP_faeces = matrix(p.adjust(uncorrectedP, "BH"), nrow=2, ncol=3)

# Milk samples
milk_featureClustering = milk_relAbs %>% select(cluster_milk_WHZ$OTU) %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint) %>% pivot_longer(-c(subject,timepoint)) %>% mutate(OTU = name) %>% select(-name) %>% left_join(cluster_milk_BMI) %>% filter(cluster %in% c(1,2)) %>% select(OTU, cluster) %>% unique()

milk_relAbs = milk_relAbs %>% mutate(subject = milk_sampleMeta$subject, timepoint = milk_sampleMeta$Timepoint)

uncorrectedP = matrix(NA, nrow=2, ncol=4)
days = c(3, 30, 60, 90)

for(i in 1:length(days)){
  day = days[i]
  for(cluster in 1:2){
    uncorrectedP[cluster, i] = ASVgroupPermutationTest_WHZ(milk_relAbs, milk_featureClustering, day, cluster, 999)[[7]]
  }
}

correctedP_milk = matrix(p.adjust(uncorrectedP, "BH"), nrow=2, ncol=4)
```

```{r Table 1 correlations of whz and bmiz}
temp=faeces_sampleMeta %>% select(whz.6m, whz.1y, whz.2y, whz.3y, BMIz.1y, BMIz.2y, BMIz.3y) %>% as.matrix()
cor(temp, use="pairwise.complete.obs")
```